packages/core/src/hooks/auth/usePermissions/index.ts

```diff
@@ -10,14 +10,16 @@ import { useKeys } from "@hooks/useKeys";
 import { useAuthBindingsContext, useLegacyAuthContext } from "@contexts/auth";
 import { PermissionResponse } from "../../../interfaces";
 
-export type UsePermissionsLegacyProps<TData = any> = {
+export type UsePermissionsLegacyProps<TData = any, TParams = unknown> = {
     v3LegacyAuthProviderCompatible: true;
     options?: UseQueryOptions<TData>;
+    params?: TParams;
 };
 
-export type UsePermissionsProps<TData = PermissionResponse> = {
+export type UsePermissionsProps<TData = PermissionResponse, TParams = unknown> = {
```

Comment:
```suggestion
export type UsePermissionsProps<TData = PermissionResponse, TParams extends Record<string, any> = Record<string, any>> = {
```

Let's make this type `any object`.